sbc_email/models/import_letters_history.py

```diff
@@ -379,28 +380,109 @@ def _get_smb_connection(self):
             return SMBConnection(
                 SmbConfig.smb_user, SmbConfig.smb_pass, 'openerp', 'nas')
 
-    @api.model
-    def import_web_letter(self):
+    @api.one
+    def import_web_letter(self, child_code, sponsor_ref, language_name,
+                          original_text, template_name, pdf_letter=None):
         """
         Call when a letter is set on web site:
             - add web letter to an import set with import letter config
               'Web letter'
         """
-        import_web_letter = self.search([
-            ('config_id.name', '=', 'Web Letter'),
-            ('state', '!=', 'done')])
+        child_id = None
+        if child_code:
+            # Retrieve child code and find corresponding id
+            model_child = self.env['compassion.child'].search(
+                [('code', '=', child_code)], limit=1)
+
+            child_id = model_child.id
+            logger.info("child id found {}".format(child_id))
+        else:
+            return False
+
+        sponsor_id = None
+        if sponsor_ref:
+            # Retrieve sponsor reference and find corresponding id
+            model_sponsor = self.env['res.partner'].search(
+                [('ref', '=', sponsor_ref)], limit=1)
```

Comment:
Add condition `is_company = False`, because sometimes company partners can have same reference as the private person.